### PR TITLE
mgr/telemetry: add `basic_pool_options_bluestore` collection

### PR DESCRIPTION
**Pull request by ljflores**
Collects per-pool bluestore options, such as `bluestore_compression_mode`.

See https://docs.ceph.com/en/latest/rados/configuration/bluestore-config-ref/#inline-compression for
where the per-pool bluestore options are documented.

Signed-off-by: Laura Flores <lflores@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
Put "nag=true" for this one since Mark was interested in collecting it, but I can set it to false if needed.

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -1086,7 +1093,17 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
                                             'wr': pool_stats['wr'],
                                             'wr_bytes': pool_stats['wr_bytes']
                         }
-
+                    # basic_pool_options_bluestore collection
```
[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/telemetry/module.py:1096]: Makes sense! Will have this fixed soon.

File: src/pybind/mgr/telemetry/module.py [hunk 2]
```diff
@@ -1086,7 +1093,17 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
...
+                    if self.is_enabled_collection(Collection.basic_pool_options_bluestore):
+                        pool_data['options'] = {'bluestore': {}}
```
[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/telemetry/module.py:1098]: There are other per-pool settings in `options` aside from bluestore compression. For instance, the gibba cluster has these pg options enabled on the `.mgr` pool:
```
"options": {
            "pg_num_max": 32,
            "pg_num_min": 1
        },
```

These are all of the available pool options:
```
osd pool set <pool> <var:size|min_size|pg_num|pgp_num|pgp_num_actual|crush_rule|hashpspool|nodelete|nopgchange|        set pool parameter <var> to <val>
 nosizechange|write_fadvise_dontneed|noscrub|nodeep-scrub|hit_set_type|hit_set_period|hit_set_count|hit_set_fpp|use_   
 gmt_hitset|target_max_bytes|target_max_objects|cache_target_dirty_ratio|cache_target_dirty_high_ratio|cache_target_   
 full_ratio|cache_min_flush_age|cache_min_evict_age|min_read_recency_for_promote|min_write_recency_for_promote|fast_   
 read|hit_set_grade_decay_rate|hit_set_search_last_n|scrub_min_interval|scrub_max_interval|deep_scrub_interval|        
 recovery_priority|recovery_op_priority|scrub_priority|compression_mode|compression_algorithm|compression_required_    
 ratio|compression_max_blob_size|compression_min_blob_size|csum_type|csum_min_block|csum_max_block|allow_ec_           
 overwrites|fingerprint_algorithm|pg_autoscale_mode|pg_autoscale_bias|pg_num_min|pg_num_max|target_size_bytes|target_  
 size_ratio|dedup_tier|dedup_chunk_algorithm|dedup_cdc_chunk_size|eio|bulk> <val> [--yes-i-really-mean-it]             
osd pool set noautoscale
```

The `bluestore` key anticipates a scenario where we want to add more per-pool options of a kind other than `bluestore` to "options". I also felt like that could help with the clarity of what the options are used for i.e.:
```
  "options": {
      "bluestore": {
           "compression_mode": "aggressive",
       },
       "pg_autoscaler": {
            "pg_num_max": 32,
            "pg_num_min": 1,
            "target_size_ratio": 1
        }
  },
```

[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/telemetry/module.py:1098]: It shows up all grouped together in the OSD map though, so maybe we should keep it uncategorized for consistency's sake. What do you think?

```
            "options": {
                "compression_mode": "aggressive",
                "pg_num_max": 32,
                "pg_num_min": 1,
                "target_size_ratio": 1
            },
```

[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/telemetry/module.py]: For now, I removed the `bluestore` key to keep it in line with how the OSD map displays options. But let me know if you think it should be added back.

[the remark this answers is not in the document]
Comment (yaarith) [reply src/pybind/mgr/telemetry/module.py]: I think there is no need to add the `bluestore` key since we wish to reflect the original data structure

**Review by yaarith — APPROVED** [review]
> Put "nag=true" for this one since Mark was interested in collecting it, but I can set it to false if needed.

I'm undecided if we should nag in this case, since we wish to have stats on these options on the one hand, but had a couple of nagging already since Quincy was released, which might be too much

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
> > Put "nag=true" for this one since Mark was interested in collecting it, but I can set it to false if needed.
> 
> I'm undecided if we should nag in this case, since we wish to have stats on these options on the one hand, but had a couple of nagging already since Quincy was released, which might be too much

Let's discuss during the Telemetry huddle.

**Comment by ljflores** [comment]
Fixed it so nag=False.

**Comment by ljflores** [comment]
Rados suite review:

https://pulpito.ceph.com/?branch=wip-yuri-testing-2022-12-06-1204
https://pulpito.ceph.com/?branch=wip-yuri-testing-2022-12-12-1136

Failures, unrelated:
    1. https://tracker.ceph.com/issues/58096
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/58173
    4. https://tracker.ceph.com/issues/52129
    5. https://tracker.ceph.com/issues/58097
    6. https://tracker.ceph.com/issues/57546
    7. https://tracker.ceph.com/issues/58098
    8. https://tracker.ceph.com/issues/57731
    9. https://tracker.ceph.com/issues/55606
    10. https://tracker.ceph.com/issues/58256
    11. https://tracker.ceph.com/issues/58258

Details:
    1. test_cluster_set_reset_user_config: NFS mount fails due to missing ceph directory - Ceph - Orchestrator
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    3. api_aio_pp: failure on LibRadosAio.SimplePoolEIOFlag and LibRadosAio.PoolEIOFlag - Ceph - RADOS
    4. LibRadosWatchNotify.AioWatchDelete failed - Ceph - RADOS
    5. qa/workunits/post-file.sh: kex_exchange_identification: read: Connection reset by peer - Ceph - RADOS
    6. rook: ensure CRDs are installed first - Ceph - Orchestrator
    7. qa/workunits/rados/test_crash.sh: crashes are never posted - Ceph - RADOS
    8. Problem: package container-selinux conflicts with udica < 0.2.6-1 provided by udica-0.2.4-1 - Infrastructure
    9. [ERR] Unhandled exception from module ''devicehealth'' while running on mgr.y: unknown - Ceph - CephSqlite
    10. ObjectStore/StoreTestSpecificAUSize.SpilloverTest/2: Expected: (logger->get(l_bluefs_slow_used_bytes)) >= (16 * 1024 * 1024), actual: 0 vs 16777216 - Ceph - Bluestore
    11. rook: kubelet fails from connection refused - Ceph - Orchestrator

**Pull request merged**